kyuubi-server/src/main/scala/org/apache/kyuubi/operation/BatchJobSubmission.scala

```diff
@@ -212,7 +215,8 @@ class BatchJobSubmission(
         metadata match {
           case Some(metadata) if metadata.peerInstanceClosed =>
             setState(OperationState.CANCELED)
-          case Some(metadata) if metadata.state == OperationState.PENDING.toString =>
+          case Some(metadata)
+              if metadata.state == OperationState.PENDING.toString && metadata.engineId == null =>
```

Comment:
In case the applicationInfo is set by peer

https://github.com/apache/kyuubi/blob/7e199d6fdbdf52222bb3eadd056b9e5a2295f36e/kyuubi-server/src/main/scala/org/apache/kyuubi/server/api/v1/BatchesResource.scala#L369-L385